docs/source/en/core/development.md

```diff
@@ -1 +1,339 @@
+### Run Test On Single File
+
+`npm test` command will look for all the files ended with `.test.js` under test folder (default [glob] matching rule is `test/**/*.test.js`).
+
+However, we would like to run only the test cases that we are working on sometimes. In this case, we may specify a file by following way:
+
+```bash
+$ TESTS=test/x.test.js npm test
+```
+
+[glob] expressions are supported here.
+
+### Set Reporter
+
+Mocha supports various reporters. Default reporter is `spec`.
+
+Reporter is allowed to be specified mannually via setting `TEST_REPORTER` environment variable. For example, to use `dot` instead:
+
+```bash
+$ TEST_REPORTER=dot npm test
+```
+
+![image](https://cloud.githubusercontent.com/assets/156269/21849809/a6fe6df8-d842-11e6-8507-20da63bc8b62.png)
+
+### Set Timeout
+
+The default timeout is 30 seconds. We may set our own timeout (in milliseconds). For example, setting timeout to 5 seconds:
+
+```bash
+$ TEST_TIMEOUT=5000 npm test
+```
+
+### Pass Parameters Via argv
+
+Besides to environment variables, `egg-bin test` also supports passing parameters directly, and it supports all mocha parameters. You may refer to [mocha usage](https://mochajs.org/#usage).
+
+```bash
+$ # Passing parameters via npm need to add an extra `--`, according to https://docs.npmjs.com/cli/run-script
+$ npm test -- --help
+$
+$ # Equivalent to `TESTS=test/**/test.js npm test`. Since the limitation of bash, it's better to add double qoute to path.
+$ npm test "test/**/test.js"
+$
+$ # Equivalent to `TEST_REPORTER=dot npm test`
+$ npm test -- --reporter=dot
+$
+$ # Parameters of mocha are supported, such as grep, require, etc.
+$ npm test -- -t 30000 --grep="should GET"
+```
+
+## Code Coverage
+
+egg-bin has build-in [nyc](https://github.com/istanbuljs/nyc) to support calculating code coverage report of unit testing.
+
+Add `npm scripts` into `package.json`：
+
+```json
+{
+  "scripts": {
+    "cov": "egg-bin cov"
+  }
+}
+```
+
+And then we can get code coverage report of unit testing via command `npm run cov`.
+
+```bash
+$ egg-bin cov
+
+  test/controller/home.test.js
+    GET /
+      ✓ should status 200 and get the body
+    POST /post
+      ✓ should status 200 and get the request body
+
+  ...
+
+  16 passing (1s)
+
+=============================== Coverage summary ===============================
+Statements   : 100% ( 41/41 )
+Branches     : 87.5% ( 7/8 )
+Functions    : 100% ( 10/10 )
+Lines        : 100% ( 41/41 )
+================================================================================
+```
+
+And we may open HTML file of complete code coverage report via command `open coverage/lcov-report/index.html`.
+
+![image](https://cloud.githubusercontent.com/assets/156269/21845201/a9a85ab6-d82c-11e6-8c24-5e85f352be4a.png)
+
+### Config Environment
+
+Same as `test` script, to run `cov`, environment need to be set to `env: unittest` as well. The configuration is also an object combined from `config.local.js` and `config.unittest.js` files.
```

Comment:
Just like `test`, the environment needs to be set to `env:unittest` to run `cov`, and the configuration comes from the combination of both `config.local.js` and `config.unittest.js`.